src/CatSearch.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react"
+import { useState, useEffect, useCallback } from "react"
+import { fetchSearchResultsFromAPI } from "./searchApiClient"
+
+export function CatSearch() {
+  const [valid, setValid] = useState(false)
+  const [hits, setHits] = useState<any[]>([])
+  const [searchTerm, setSearchTerm] = useState("")
+
+  const searchTermOnChange = useCallback((event) => {
+    const { value } = event.target
+    setSearchTerm(value)
+  }, [])
+
+  useEffect(() => {
+    setValid(true)
+    if (searchTerm.length < 2) {
+      setValid(false)
```

Comment:
If we have an invalid search term, we should also reset the `hits` value to an empty array using `setHits([])`